egs/wsj/s5/steps/nnet3/train_discriminative.sh

```diff
@@ -0,0 +1,376 @@
+        fi
+
+        #archive=$[(($n+($x*$num_jobs_nnet))%$num_archives)+1]
+        if $scale_max_param_change; then
+          this_max_param_change=$(perl -e "print ($max_param_change * $num_jobs_nnet);")
+        else
+          this_max_param_change=$max_param_change
+        fi
+
+        $cmd $train_queue_opt $dir/log/train.$x.$n.log \
+          nnet3-discriminative-train \
+          --apply-deriv-weights=$apply_deriv_weights \
+          $parallel_train_opts $deriv_time_opts \
+          --max-param-change=$this_max_param_change \
+          --silence-phones=$silphonelist \
+          --criterion=$criterion --drop-frames=$drop_frames \
+          --one-silence-class=$one_silence_class \
+          --boost=$boost --acoustic-scale=$acoustic_scale $regularization_opts \
+          $dir/$x.mdl \
+          "ark:nnet3-discriminative-copy-egs --frame-shift=$frame_shift --truncate-deriv-weights=$truncate_deriv_weights ark:$degs_dir/degs.$archive.ark ark:- | nnet3-discriminative-shuffle-egs --buffer-size=$shuffle_buffer_size --srand=$x ark:- ark:- | nnet3-discriminative-merge-egs --minibatch-size=$minibatch_size ark:- ark:- |" \
+          $dir/$[$x+1].$n.raw || touch $dir/.error &
+      done
+      wait
+      [ -f $dir/.error ] && exit 1
+    )
+    [ -f $dir/.error ] && { echo "Found $dir/.error. See $dir/log/train.$x.*.log"; exit 1; }
+
+    nnets_list=$(for n in $(seq $num_jobs_nnet); do echo $dir/$[$x+1].$n.raw; done)
+
+    # below use run.pl instead of a generic $cmd for these very quick stages,
+    # so that we don't run the risk of waiting for a possibly hard-to-get GPU.
+    run.pl $dir/log/average.$x.log \
+      nnet3-average $nnets_list - \| \
+      nnet3-am-copy --set-raw-nnet=- $dir/$x.mdl $dir/$[$x+1].mdl || exit 1;
+
+    if $modify_learning_rates; then
+      run.pl $dir/log/modify_learning_rates.$x.log \
+        nnet3-modify-learning-rates --retroactive=$retroactive \
+        --last-layer-factor=$last_layer_factor \
+        --first-layer-factor=$first_layer_factor \
+        "nnet3-am-copy --raw $dir/$x.mdl -|" "nnet3-am-copy --raw $dir/$[$x+1].mdl -|" - \| \
+        nnet3-am-copy --set-raw-nnet=- $dir/$x.mdl $dir/$[$x+1].mdl || exit 1;
+    fi
+    rm $nnets_list
+
+    if [ ! -z "${iter_to_epoch[$x]}" ]; then
+      e=${iter_to_epoch[$x]}
+      ln -sf $x.mdl $dir/epoch$e.mdl
+    fi
+
+    if $adjust_priors && [ ! -z "${iter_to_epoch[$x]}" ]; then
+      if [ ! -f $degs_dir/priors_egs.1.ark ]; then
+        echo "$0: Expecting $degs_dir/priors_egs.1.ark to exist since --adjust-priors was true."
+        echo "$0: Run this script with --adjust-priors false to not adjust priors"
+        exit 1
+      fi
+      (
+      e=${iter_to_epoch[$x]}
+      rm -f $dir/.error 2> /dev/null || true
+
+      steps/nnet3/adjust_priors.sh --egs-type priors_egs \
+        --num-jobs-compute-prior $num_archives_priors \
+        --cmd "$cmd $prior_queue_opt" --use-gpu false \
```

Comment:
passing in options like this to control allocation of the GPU is deprecated in favor of the new-style options-- please remove it, and instead have adjust_priors.sh itself automatically add "--gpu 1" to any command that needs a GPU, if it is directed to use the GPU.  BTW, this line would have a bug anyway as you hardcode --use-gpu false to adjust_priors.sh.